src/System.Reflection.Emit.Lightweight/tests/DynamicILInfoTests.cs

```diff
@@ -0,0 +1,62 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Xunit;
+
+namespace System.Reflection.Emit.Tests
+{
+    public class DynamicILInfoTests
+    {
+        private const string FieldName = "_id";
+
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public unsafe void SetX_NullInput_ThrowsArgumentNullException(bool skipVisibility)
+        {
+            DynamicMethod method = GetDynamicMethod(skipVisibility);
+            DynamicILInfo dynamicILInfo = method.GetDynamicILInfo();
+
+            Assert.Throws<ArgumentNullException>(() => dynamicILInfo.SetCode(null, 1, 8));
+            Assert.Throws<ArgumentNullException>(() => dynamicILInfo.SetExceptions(null, 1));
+            Assert.Throws<ArgumentNullException>(() => dynamicILInfo.SetLocalSignature(null, 1));
+        }
+
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public unsafe void SetX_NegativeInputSize_ThrowsArgumentOutOfRangeException(bool skipVisibility)
+        {
+            DynamicMethod method = GetDynamicMethod(skipVisibility);
+            DynamicILInfo dynamicILInfo = method.GetDynamicILInfo();
+            var bytes = new byte[] { 0x02, 0x02, 0x02, 0x02, 0x02, 0x02, 0x02, 0x02, 0x02, 0x02 };
+
+            Assert.Throws<ArgumentOutOfRangeException>(() => {fixed (byte* bytesPtr = bytes) { dynamicILInfo.SetCode(bytesPtr, -1, 8); }});
+            Assert.Throws<ArgumentOutOfRangeException>(() => {fixed (byte* bytesPtr = bytes) { dynamicILInfo.SetExceptions(bytesPtr, -1); }});
+            Assert.Throws<ArgumentOutOfRangeException>(() => {fixed (byte* bytesPtr = bytes) { dynamicILInfo.SetLocalSignature(bytesPtr, -1); }});
+        }
+
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public void GetDynamicILInfo_IsUniqueNotNull(bool skipVisibility)
+        {
+            DynamicMethod method = GetDynamicMethod(skipVisibility);
+            DynamicILInfo dynamicILInfo = method.GetDynamicILInfo();
+            
+            Assert.NotNull(dynamicILInfo);
+            Assert.Equal(dynamicILInfo, method.GetDynamicILInfo());
+            Assert.Equal(method, dynamicILInfo.DynamicMethod);
+        }
+        
```

Comment:
You should have a test that DynamicILInfo.DynamicMethod has actual code in it. I don't think any of your tests do this.
For example, dynamically generate a method that returns 42, get the DynamicMethod, then invoke it and check you get 42.